src/Pennant/FeatureDirective.php

```diff
@@ -0,0 +1,63 @@
+<?php declare(strict_types=1);
+
+namespace Nuwave\Lighthouse\Pennant;
+
+use Laravel\Pennant\FeatureManager;
+use Nuwave\Lighthouse\Exceptions\DefinitionException;
+use Nuwave\Lighthouse\Schema\Directives\HideDirective;
+
+final class FeatureDirective extends HideDirective
+{
+    public function __construct(
+        private FeatureManager $features,
+    ) {
+        parent::__construct();
+    }
+
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+"""
+Conditionally add the annotated element to schema depending of the state of a feature using Laravel Pennant.
+"""
+directive @feature(
+    """
+    The name of the feature to be checked (can be a string or class name).
+    """
+    name: String
+    
+    """
+    Specify what the state of the feature should be for the field to be applied.
+    """
+    when: FeatureState = ACTIVE
```

Comment:
```suggestion
    when: FeatureState! = ACTIVE
```

Optional, but non-null